pkg/api/types.go

```diff
@@ -165,6 +171,13 @@ const (
 	TerminationMessagePathDefault string = "/dev/termination-log"
 )
 
+// ClusterReference contains information to identify a cluster.
+type ClusterReference struct {
+	// The name of the cluster.
+	// Cannot be updated.
```

Comment:
If the name is immutable, then it should be an ID.

I also think that this needs a ClusterUID field.